api/src/main/java/org/openmrs/CohortMembership.java

```
@@ -101,4 +115,16 @@ public void setEndDate(Date endDate) {
 	public int compareTo(CohortMembership o) {
 		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
 	}
+	
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof CohortMembership) {
+			CohortMembership o = (CohortMembership) obj;
+			return Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId()) &&
+					Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId()) &&
+					this.getStartDate() == o.getStartDate() &&
+					this.getEndDate() == o.getEndDate();
+		}
+		return super.equals(obj);
```

this should throw an exception since it isn't of the same type